lib/bin/lwm2m_carrier/lwm2m_carrier.rst

```diff
@@ -3,24 +3,96 @@
+
+.. note::
+   A change of the bootstrap server URI between builds does not trigger a new bootstrap.
+   The bootstrap process is intended to happen only once, unless it is initiated from the server.
+   To redo the bootstrap process, you must erase the flash and then load your application again.
+
+For a production build, the :c:func:`lwm2m_carrier_init` function should always be initialized without parameters.
+After calling the :cpp:func:`lwm2m_carrier_init` function, your application can call the :cpp:func:`lwm2m_carrier_run` non-returning function in its own thread.
+Both these functions are called in :file:`nrf\\lib\\bin\\lwm2m_carrier\\os\\lwm2m_carrier.c`, which is included into the project when you enable the LwM2M carrier library.
+
+The :cpp:func:`lwm2m_carrier_event_handler` function must be implemented by your application.
+This is shown in the :ref:`lwm2m_carrier` sample.
+A basic implementation is included in :file:`nrf\\lib\\bin\\lwm2m_carrier\\os\\lwm2m_carrier.c`.
+
+See :file:`nrf\\lib\\bin\\lwm2m_carrier\\include\\lwm2m_carrier.h` for all the events and API.
+
+
+LwM2M carrier library events
+============================
+
+:c:macro:`LWM2M_CARRIER_EVENT_BSDLIB_INIT`
+   This event indicates that the :ref:`bsdlib` is initialized and can be used.
+   (See :ref:`req_appln_limitations`).
+
+:c:macro:`LWM2M_CARRIER_EVENT_CONNECTING`, :c:macro:`LWM2M_CARRIER_EVENT_CONNECTED`, :c:macro:`LWM2M_CARRIER_EVENT_DISCONNECTING`, :c:macro:`LWM2M_CARRIER_EVENT_DISCONNECTED`
+   These events indicate connecting to or disconnecting from the LTE network.
+   They occur during the bootstrapping process, startup, and during FOTA.
+
+:c:macro:`LWM2M_CARRIER_EVENT_BOOTSTRAPPED`
+   This event indicates that the bootstrap sequence is complete.
+   This event is typically seen during the first boot-up.
+   It also indicates that the device is ready to be registered.
+
+:c:macro:`LWM2M_CARRIER_EVENT_READY`
+   This event indicates that the device has connected successfully to the operator's device management server.
+
+:c:macro:`LWM2M_CARRIER_EVENT_DEFERRED`
+   If the connection to the device management server fails, the LwM2M carrier library must wait for a duration of 24 hours before retrying.
+   In this case, the :c:macro:`LWM2M_CARRIER_EVENT_DEFERRED` event appears instead of the :c:macro:`LWM2M_CARRIER_EVENT_READY` event.
+   The retry results in a :c:macro:`LWM2M_CARRIER_EVENT_READY` event, or another :c:macro:`LWM2M_CARRIER_EVENT_DEFERRED` event.
+
+:c:macro:`LWM2M_CARRIER_EVENT_FOTA_START`
+   This event indicates that modem update has started.
+   The application should not start any new TLS sessions.
+   See :ref:`req_appln_limitations`.
+
+:c:macro:`LWM2M_CARRIER_EVENT_REBOOT`
+   If the application is not ready to reboot, this event returns non-zero and then reboots at the earliest convenient time.
+
+:c:macro:`LWM2M_CARRIER_EVENT_ERROR`
```

Comment:
This section needs to be improved to show which error codes is given for which failure category, and what the error codes indicates.